File: sentence.py
# Ellie Kozlowski
# Solution to sentence
# 02/10/19

n = int(input())

# loop through queries
for i in range(n):
    m = int(input())
    sentence = input()

    # loops through sentence and counts number of upper and lower case letters
    numUpper = sum(1 for c in sentence if c.isupper())
    numLower = sum(1 for c in sentence if c.islower())

    # checks to see if sentence is balanced
    if numUpper == numLower:
        print("Sentence #",(i + 1),": Yes", sep='')
    else:
        print("Sentence #",(i + 1),": No", sep='')
    
    

File: sentence.cpp
#include <cctype>
#include <iostream>

int main()
{
    int cases;

    std::cin >> cases;

    for (int cas = 1; cas <= cases; cas++)
    {
	int n;
	std::cin >> n;

	int upper = 0, lower = 0;
	while (n--)
	{
	    std::string word {};
	    std::cin >> word;

	    for (char ch : word)
	    {
		if (isupper(ch))
		{
		    upper++;
		}
		else 
		{
		    lower++;
		}
	    }
	}

	std::cout
	    << "Sentence #"
	    << cas
	    << ": "
	    << (upper == lower ? "Yes" : "No")
	    << std::endl;
    }    
    return 0;
}
